### PR TITLE
Correct the branches used by the release to build.

### PR DESCRIPTION
**Pull request by awholegunch**
Running as [Quick Build 2](http://builds.cask.co/browse/CDAP-DQB58-2)

### CODE REVIEW

File: cdap-docs/vars [hunk 1]
```diff
@@ -27,11 +27,12 @@ BUILD_RST_HASH_LOCATION="cdap-docs/vars"
...
+GIT_BRANCH_PARENT="release/3.5"
 
 # Used by cdap-apps for Hydrator documentation
-GIT_BRANCH_CASK_HYDRATOR="develop"
+GIT_BRANCH_CASK_HYDRATOR="release/1.4"
+GIT_BRANCH_CASK_TRACKER="release/0.2"
 
 # Used by developers-manual
-CDAP_CLIENTS_RELEASE_VERSION="1.2.0"
+CDAP_CLIENTS_RELEASE_VERSION="1.3.0"
```
Comment (anwar6953) [inline cdap-docs/vars]: Is this right? We released a 1.3 for those?

Comment (wolf31o2) [inline cdap-docs/vars]: We have not. In fact, not having a 1.3 release of the Ruby Gems makes it unusable via Chef.

Comment (anwar6953) [inline cdap-docs/vars]: You're saying that any version other than "1.3" would make it unusable?
How is it working with 1.2 right now then?

Comment (wolf31o2) [inline cdap-docs/vars]: No. I am saying that we converted it to a Gem for release to Rubygems after 1.2 and we only have a 1.3.0.a pre-release on Rubygems, which is the latest, which isn't usable in Chef. Let's just ignore this. I was simply confirming that there isn't a 1.3 release (yet).

Comment (anwar6953) [inline cdap-docs/vars]: I see. So @awholegunch should revert this change on line 36/37.

**Comment by awholegunch** [comment]
I'm waiting for @sagarkapare to create the branches in the referenced repos, and will update this PR accordingly to what the final outcome is.

**Comment by anwar6953** [comment]
@awholegunch, which repos are you referring to?

**Comment by awholegunch** [comment]
We need `release/cdap-3.5-compatible` branches created in:
- https://github.com/caskdata/cdap-apps
- https://github.com/caskdata/cdap-packs
- https://github.com/cdap-guides

I know, it's a lot of additional branches. The release manager usually does this, but until it's done, I build the docs against `develop`.

**Comment by awholegunch** [comment]
It's possible that some (or all of these) might not get changed or bumped; that's fine. Just need to let  me know which ones to build against. That's why it's on hold.

**Comment by anwar6953** [comment]
bumping https://github.com/caskdata/cdap/pull/6189#discussion_r72128257

**Comment by awholegunch** [comment]
"On-hold", as this PR may be subsumed by https://github.com/caskdata/cdap/pull/6262. 6262 has the same changes, but there are so many changes in the Hydrator repo that this branch will not build cleanly as-is.

**Comment by awholegunch** [comment]
Closing as subsumed by https://github.com/caskdata/cdap/pull/6262.
